CHANGELOG.md

```diff
@@ -2,6 +2,12 @@
 
 All notable changes to this project will be documented in this file.
 
+## [4.1.0] - 2023-08-08
```

Comment:
Why is it a minor? To my understanding, it should be a patch release if the API is not modified of a major if the API is modified.

Reply:
the API has changed in the way that upsert now returns a value.
but all previous code is still compatible so there is no breaking change per say.

Reply:
to me it can be considered a minor since it adds a feature with no breaking change regarding previous code